src/neo4j_genai/experimental/pipeline/pipeline.py

```diff
@@ -59,26 +61,28 @@ class RunResult(BaseModel):
     timestamp: datetime = Field(default_factory=datetime.utcnow)
 
 
+# Type used for annotating the partial on_task_complete parameter
+# and prevent type checking error when using kwargs
+class PartialOnTaskCompleteProtocol(Protocol):
+    async def __call__(self, task: TaskPipelineNode, res: RunResult) -> None: ...
```

Comment:
Is this complete?

Reply:
Yes, no implementation is needed for this method, it's really just used to define a "callable type" with proper kwargs.